ddtrace/tracer/spancontext.go

```diff
@@ -236,6 +236,12 @@ func (t *trace) finishedOne(s *span) {
 	if len(t.spans) != t.finished {
 		return
 	}
+
+	// Adjust the starting trace size to avoid allocations.
+	if uint32(len(t.spans)) > atomic.LoadUint32(&traceStartSize) {
```

Comment:
This is kinda ugly and repetitive, can we maybe:
```go
if n := uint32(len(t.spans)); n > atomic.LoadUint32(&traceStartSize) {
    // adjust the starting trace size to avoid repeated calls to runtime.growslice
    atomic.StoreUint32(&traceStartSize, n)
}
```